test_rmw_implementation/test/test_unique_identifiers.cpp

```diff
@@ -221,3 +262,52 @@ TEST_F(CLASSNAME(TestUniqueIdentifiersForMultiplePublishers, RMW_IMPLEMENTATION)
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_FALSE(are_equal);
 }
+
+
+class CLASSNAME (TestUniqueIdentifiersForMultipleClients, RMW_IMPLEMENTATION)
+  : public CLASSNAME(TestUniqueIdentifierAPI, RMW_IMPLEMENTATION)
+{
+protected:
+  using Base = CLASSNAME(TestUniqueIdentifierAPI, RMW_IMPLEMENTATION);
+  void SetUp() override
+  {
+    Base::SetUp();
+    rmw_publisher_options_t options = rmw_get_default_publisher_options();
+    constexpr char service_name[] = "service_name";
+    cli1 = rmw_create_client(
+      node, srv_ts, service_name, &qos_profile);
+    ASSERT_NE(nullptr, cli1) << rmw_get_error_string().str;
+    cli2 = rmw_create_client(
+      node, srv_ts, service_name, &qos_profile);
+    ASSERT_NE(nullptr, cli2) << rmw_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    rmw_ret_t ret = rmw_destroy_client(node, cli1);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+    ret = rmw_destroy_client(node, cli2);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+    Base::TearDown();
+  }
+
+  rmw_client_t * cli1 {nullptr};
+  rmw_client_t * cli2 {nullptr};
+};
+
+
+TEST_F(CLASSNAME(TestUniqueIdentifiersForMultipleClients, RMW_IMPLEMENTATION), different_clis) {
+  rmw_gid_t cli1_gid{};
+  rmw_gid_t cli2_gid{};
+
+  rmw_ret_t ret = rmw_get_gid_for_client(cli1, &cli1_gid);
+  ASSERT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+
+  ret = rmw_get_gid_for_client(cli2, &cli2_gid);
+  ASSERT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+
+  bool are_equal = true;
+  ret = rmw_compare_gids_equal(&cli1_gid, &cli2_gid, &are_equal);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_FALSE(are_equal);
+}
```

Comment:
I'm not sure this is something we should we testing here, but it's okay to add it

Reply:
Yeah, it seems a bit odd to me too. I believe this was added to copy the test for the publisher gid in this file.

Reply:
Yeah, these methods (and the comment from the previous review) were just copied over to be symmetrical with the publisher gid tests